app/uk/gov/hmrc/customerprofile/services/CustomerProfileService.scala

```diff
@@ -112,37 +114,38 @@ class CustomerProfileService @Inject() (
       } yield response
     }
 
-  private def mapStatusToExistingValue(statusReceived: Option[Status]): Option[StatusName] = statusReceived match {
-    case Some(Status.Pending)  => Some(StatusName.Pending)
-    case Some(Status.Bounced)  => Some(StatusName.Bounced)
-    case Some(Status.Verified) => Some(StatusName.Verified)
-    case None                  => None
-  }
+  def reOptInEnabledCheck(preferences: Preference): Preference =
+    if (reOptInEnabled)
+      preferences
+    else {
+      preferences.status.map(_.name) match {
+        case Some(statusName) =>
+          if (statusName.contains(ReOptIn))
+            preferences.copy(email = preferences.email.map(_.copy(status = Verified)),status = preferences.status.map(_.copy(name = Some(Verified))))
+          else preferences
+        case _ => preferences
```

Comment:
As the else is the same as the `case _` just put the if condition on the first case statement.